src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+                log.error("Unable to process ARM create record response file {}", createRecordFilenameProcessor.getCreateRecordFilenameAndPath());
+            }
+        }
+    }
+
+    private void readCreateRecordFile(BinaryData createRecordBinary, CreateRecordFilenameProcessor createRecordFilenameProcessor,
+                                      ArmBatchResponses armBatchResponses) {
+        if (nonNull(createRecordBinary)) {
+            Path jsonPath = null;
+            try {
+                boolean appendUuidToWorkspace = true;
+                jsonPath = fileOperationService.saveBinaryDataToSpecifiedWorkspace(
+                    createRecordBinary,
+                    createRecordFilenameProcessor.getCreateRecordFilename(),
+                    armDataManagementConfiguration.getTempBlobWorkspace(),
+                    appendUuidToWorkspace
+                );
+
+                if (nonNull(jsonPath) && jsonPath.toFile().exists()) {
+                    logResponseFileContents(jsonPath);
+                    ArmResponseCreateRecord armResponseCreateRecord = objectMapper.readValue(jsonPath.toFile(), ArmResponseCreateRecord.class);
+                    UploadNewFileRecord uploadNewFileRecord = readInputJson(armResponseCreateRecord.getInput());
+                    if (nonNull(uploadNewFileRecord)) {
+                        armBatchResponses.addResponseBatchData(Integer.valueOf(uploadNewFileRecord.getRelationId()),
+                                                               armResponseCreateRecord, createRecordFilenameProcessor);
+                    } else {
+                        log.warn("Failed to obtain relation id from create record");
+                    }
+                } else {
+                    log.warn("Failed to write create record file to temp workspace {}", createRecordFilenameProcessor.getCreateRecordFilenameAndPath());
+                }
+            } catch (IOException e) {
+                log.error("Unable to write create record file to temporary workspace {} - {}", createRecordFilenameProcessor.getCreateRecordFilenameAndPath(),
+                          e.getMessage());
+            } catch (Exception e) {
+                log.error("Unable to process arm response create record file {}", createRecordFilenameProcessor.getCreateRecordFilenameAndPath(), e);
+            } finally {
+                cleanupTemporaryJsonFile(jsonPath);
+            }
+        } else {
+            log.warn("Failed to read create record file {}", createRecordFilenameProcessor.getCreateRecordFilenameAndPath());
+        }
+
+    }
+
+    private void processUploadResponseFiles(List<UploadFileFilenameProcessor> uploadFileResponses, ArmBatchResponses armBatchResponses) {
+        for (UploadFileFilenameProcessor uploadFileFilenameProcessor : uploadFileResponses) {
+            try {
+                BinaryData uploadFileBinary = armDataManagementApi.getBlobData(uploadFileFilenameProcessor.getUploadFileFilenameAndPath());
```

Comment:
Same

Reply:
changes to existing code that are not necessary for plugging-in DETS functionality cannot be tackled in this PR, as explained in the PR description